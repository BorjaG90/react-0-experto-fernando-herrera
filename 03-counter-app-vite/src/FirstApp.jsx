import PropTypes from 'prop-types';

const newMessage = {
  message: 'Hola Mundo',
  title: 'Borja'
};

/* Si no hay una dependencia con el 
comportamiento interno del componente,
es mejor sacar las funciones fuera del componente
(en este caso FirstApp)
*/
const getResult = (a,b) => {
  return a + b
}

export const FirstApp = ({title, subTitle, name}) => {
  return (
    <>
      <h2>{title}</h2>
      {/* <code>{ JSON.stringify(newMessage) }</code> */}
      {/* <h3>Soy un subtítulo</h3> */}
      {/* <h3>{ getResult(3,5) }</h3> */}
      <h3>{subTitle}</h3>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name: 'Borja Gete',
  subTitle: 'No hay subtítulo',
  // title: 'No hay título',
}
