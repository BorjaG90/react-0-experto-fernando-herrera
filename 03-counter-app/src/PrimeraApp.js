// Functional Components
import React from "react";
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo}) => {
  if (!saludo) {
    throw new Error('El saludo es necesario');
  }
  return (
    <>
      <h1>{saludo}</h1>
      <p>Mi primera aplicaci√≥n</p>
    </>
  );
}

// Definicion de las props del componente
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

export default PrimeraApp;