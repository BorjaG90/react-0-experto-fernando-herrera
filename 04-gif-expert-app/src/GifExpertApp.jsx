import { useState } from "react"
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['OnePunch']);

  const onAddCategory = (newCategory) => {
    if(categories.includes(newCategory)) return;
    
    setCategories([newCategory, ...categories]);
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory
        onNewCategory={onAddCategory}
      />
      {
        categories.map((category) => {
          return (
            <div key={category}>
              <GifGrid key={category} category={category}/>
            </div>
          )
        })
      }
    </>
  )
}
