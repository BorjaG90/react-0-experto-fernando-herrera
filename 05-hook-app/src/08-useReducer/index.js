export * from './TodoApp';
export * from './TodoList';
export * from './TodoAdd';
export * from './TodoItem';