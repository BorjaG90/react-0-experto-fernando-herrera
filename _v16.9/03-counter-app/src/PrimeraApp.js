// Functional Components
import React from "react";
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, subtitulo}) => {
  if (!saludo) {
    throw new Error('El saludo es necesario');
  }
  return (
    <>
      <h1>{saludo}</h1>
      <p>{subtitulo}</p>
    </>
  );
}

// Definicion de las props del componente
PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;