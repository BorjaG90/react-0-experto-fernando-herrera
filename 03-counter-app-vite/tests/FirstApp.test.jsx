import {render} from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => { 
  test('debe de hacer match con el snapshot', () => {
    const title = 'Hola, soy Goku';
    const {container} = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el tÃ­tulo en un h2', () => {
    const title = 'Hola, soy Goku';
    const {container, getByText} = render(<FirstApp title={title} />);

    expect(getByText(title)).toBeTruthy();
    const h2 = container.querySelector('h2');
    expect(h2.innerHTML).toContain(title);
  });
});