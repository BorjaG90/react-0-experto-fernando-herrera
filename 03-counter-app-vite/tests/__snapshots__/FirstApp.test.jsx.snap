// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h2>
    Hola, soy Goku
  </h2>
  <h3>
    No hay subt√≠tulo
  </h3>
  <p>
    Borja Gete
  </p>
</div>
`;
